BraintreeCore/src/test/java/com/braintreepayments/api/BraintreeClientUnitTest.kt

```diff
@@ -639,17 +620,19 @@ class BraintreeClientUnitTest {
     fun reportCrash_reportsCrashViaAnalyticsClient() {
         every { authorizationLoader.authorizationFromCache } returns authorization
 
-        val configuration = Configuration.fromJson(Fixtures.CONFIGURATION_WITH_ANALYTICS)
+        val configuration = Configuration.fromJson(Fixtures.CONFIGURATION_WITH_ENVIRONMENT)
         val configurationLoader = MockkConfigurationLoaderBuilder()
             .configuration(configuration)
             .build()
         val params = createDefaultParams(configurationLoader, authorizationLoader)
         val sut = BraintreeClient(params)
         sut.reportCrash()
 
+        // TODO: - How to make this test wait for the getConfiguration() callback to load first
         verify {
             analyticsClient.reportCrash(
                 applicationContext,
+                any(),
                 "session-id",
                 IntegrationType.CUSTOM,
                 authorization
```

Comment:
👋 I need some help from our Android pros on how to properly test `BraintreeClient.reportCrash()` (see source code [here](https://github.com/braintree/braintree_android/blob/aad4455d68263a33f4d8c761672e04c03a0c489c/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt#L431-L441)).

Is there a way with MockK to easily mock the callback from `getConfiguration()` [here](https://github.com/braintree/braintree_android/blob/aad4455d68263a33f4d8c761672e04c03a0c489c/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt#L185)? I couldn't get anything working

Reply:
(Note this test is currently failing)

Reply:
It's not pretty, but I got the test working (committed the code). It'll get cleaner when we remove the auth call when #830 is merged.

Reply:
I see your changes, thank you!

It's seems a little weird to me to do that config fetch when trying to report an app crash (in case resources are limited) but that's the only way we'll get merchantID and env, which seem pretty critical to have in the log. We _could_ not do that config fetch and pass `null` for the config (which again would result in no env or merchantID param in the FPTI payload).

We can leave as-is, but just wanted to call it out in case you felt strongly

Reply:
Yeah the analytics without merchantID don't seem very valuable so probably fine to leave as is